File: graphene-writer/src/main/java/net/iponweb/disthene/util/CassandraLoadBalancingPolicies.java
package net.iponweb.disthene.util;

import com.datastax.driver.core.policies.*;

public class CassandraLoadBalancingPolicies {

    public static final String tokenAwarePolicy = "TokenAwarePolicy";
    public static final String tokenDcAwareRoundRobinPolicy = "TokenDcAwareRoundRobinPolicy";
    public static final String tokenLatencyAwarePolicy = "TokenLatencyAwarePolicy";

    public static LoadBalancingPolicy getLoadBalancingPolicy(String policy) {
        switch (policy) {
            case tokenAwarePolicy:
                return new TokenAwarePolicy(new RoundRobinPolicy());
            case tokenDcAwareRoundRobinPolicy:
                return new TokenAwarePolicy(DCAwareRoundRobinPolicy.builder().build());
            case tokenLatencyAwarePolicy:
                return new TokenAwarePolicy(LatencyAwarePolicy.builder(new RoundRobinPolicy()).build());
            default:
                throw new IllegalArgumentException("Cassandra load balancing policy can be " + tokenAwarePolicy + " ," + tokenLatencyAwarePolicy
                       + " ," + tokenDcAwareRoundRobinPolicy);
        }
    }
}

File: graphene-writer/src/main/kotlin/com/graphene/writer/store/data/CassandraLoadBalancingPolicies.kt
package com.graphene.writer.store.data

import com.datastax.driver.core.policies.*

object CassandraLoadBalancingPolicies {

  val tokenAwarePolicy = "TokenAwarePolicy"
  val tokenDcAwareRoundRobinPolicy = "TokenDcAwareRoundRobinPolicy"
  val tokenLatencyAwarePolicy = "TokenLatencyAwarePolicy"

  fun getLoadBalancingPolicy(policy: String): LoadBalancingPolicy {
    when (policy) {
      tokenAwarePolicy -> return TokenAwarePolicy(RoundRobinPolicy())
      tokenDcAwareRoundRobinPolicy -> return TokenAwarePolicy(DCAwareRoundRobinPolicy.builder().build())
      tokenLatencyAwarePolicy -> return TokenAwarePolicy(LatencyAwarePolicy.builder(RoundRobinPolicy()).build())
      else -> throw IllegalArgumentException("Cassandra load balancing policy can be " + tokenAwarePolicy + " ," + tokenLatencyAwarePolicy
        + " ," + tokenDcAwareRoundRobinPolicy)
    }
  }
}
